### ISSUE TITLE
Some questions about grad-CAM showing in fig7 in paper Tag2Text.

### ISSUE THREAD

**Issue by SKBL5694** [xinyu1205/recognize-anything#70]
I use gradCAM to visualize the same image as the paper, but I get a weird result different from the fig7.
When I use the word "cat" to calculate the heatmap: the result is like this:
![image](https://github.com/xinyu1205/recognize-anything/assets/42370333/feca9755-4560-4ac7-9745-6b72996e56c4)
when I change the word to "siamese"(a kind of cat), the result looks ok
![image](https://github.com/xinyu1205/recognize-anything/assets/42370333/812a24b1-b893-4efd-80b6-349b0883dddd)
sorry, when I raise this issue, I find I use the weights of ram rather T2T, however, the gradCAM fig is in the T2T paper. But it still seems weird, can you tell me what caused this?
I don't remember whether swin-transformer is also used in t2t. If it is not used in t2t, I suspect it is caused by this; other than that, all I can think of is CLIP text encoder and text encoder trained by yourself in T2T.

**Comment by xinyu1205** [comment]
Did you use the image-tag recognition decoder on the tagging task to obtain the Grad-Cam? Since the figure 7 of Tag2Text is obtained based on the backward gradient of the image-tag interaction encoder on the generation task.
And I also found that the image-tag recognition decoder's Grad-Cam is often a meaningless scatter graph, even if it predicts high logits. Normally, with good recognition performance, its grad-cam should be very accurate. I haven't found the reason yet.

**Comment by SKBL5694** [comment]
It seems that I am indeed doing gradcam on the recognition task, because your code does not open the generation task for RAM, but I have added the generation task to RAM in the way of T2T, I will test it, thank you.
Also, I share your opinion that "with good recognition performance, its grad-cam should be very accurate". But I also encountered a similar situation [pytorch-grad-cam/issues/84](https://github.com/jacobgil/pytorch-grad-cam/issues/84#issuecomment-904522521) in some other swin-transformer-based discussions about gradcam, but unfortunately these discussions were fruitless. I think it may be due to the patch merging operation in swin-transformer that the features lose the meaning of the traditional spatial structure, but this can't explain why gradcam is accurate sometimes, and I am also very confused. Thanks for your reply though, I'll try it again. Thank you again for your excellent work and kind reply.

**Comment by xinyu1205** [comment]
Thank you for your interest and your kind words, welcome to provide feedback if you have more issues.

**Comment by pribadihcr** [comment]
Hi @SKBL5694 , Have you resolve it?
